File: core/bench/report/src/types/benchmark_kind.rs
/* Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

use derive_more::Display;
use serde::{Deserialize, Serialize};

#[derive(
    Debug,
    Clone,
    Copy,
    PartialEq,
    Eq,
    Hash,
    Display,
    Serialize,
    Deserialize,
    Default,
    PartialOrd,
    Ord,
)]
pub enum BenchmarkKind {
    #[display("Pinned Producer")]
    #[serde(rename = "pinned_producer")]
    PinnedProducer,
    #[display("Pinned Consumer")]
    #[serde(rename = "pinned_consumer")]
    #[default]
    PinnedConsumer,
    #[display("Pinned Producer And Consumer")]
    #[serde(rename = "pinned_producer_and_consumer")]
    PinnedProducerAndConsumer,
    #[display("Balanced Producer")]
    #[serde(rename = "balanced_producer")]
    BalancedProducer,
    #[display("Balanced Consumer Group")]
    #[serde(rename = "balanced_consumer_group")]
    BalancedConsumerGroup,
    #[display("Balanced Producer And Consumer Group")]
    #[serde(rename = "balanced_producer_and_consumer_group")]
    BalancedProducerAndConsumerGroup,
    #[display("End To End Producing Consumer")]
    #[serde(rename = "end_to_end_producing_consumer")]
    EndToEndProducingConsumer,
    #[display("End To End Producing Consumer Group")]
    #[serde(rename = "end_to_end_producing_consumer_group")]
    EndToEndProducingConsumerGroup,
}

File: foreign/java/bench/src/main/java/org/apache/iggy/bench/common/enums/BenchmarkKind.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.apache.iggy.bench.common.enums;

public enum BenchmarkKind {
    PINNED_PRODUCER("pinned_producer", "Pinned Producer"),
    PINNED_CONSUMER("pinned_consumer", "Pinned Consumer"),
    PINNED_PRODUCER_AND_CONSUMER("pinned_producer_and_consumer", "Pinned Producer And Consumer"),
    BALANCED_PRODUCER("balanced_producer", "Balanced Producer"),
    BALANCED_CONSUMER_GROUP("balanced_consumer_group", "Balanced Consumer Group"),
    BALANCED_PRODUCER_AND_CONSUMER_GROUP(
            "balanced_producer_and_consumer_group", "Balanced Producer And Consumer Group"),
    END_TO_END_PRODUCING_CONSUMER("end_to_end_producing_consumer", "End To End Producing Consumer"),
    END_TO_END_PRODUCING_CONSUMER_GROUP("end_to_end_producing_consumer_group", "End To End Producing Consumer Group");

    private final String value;
    private final String prettyName;

    BenchmarkKind(String value, String prettyName) {
        this.value = value;
        this.prettyName = prettyName;
    }

    public String value() {
        return value;
    }

    public String prettyName() {
        return prettyName;
    }

    @Override
    public String toString() {
        return value;
    }
}
